src/components/WebexActivityStream/WebexActivityStream.js

```diff
@@ -138,19 +170,34 @@ Greeting.propTypes = {
   personName: PropTypes.string.isRequired,
 };
 
+export function TimeRuler({text}) {
+  return <ListSeparator className="time-ruler" role="listitem" text={text} />;
+}
+
+TimeRuler.propTypes = {
+  text: PropTypes.string.isRequired,
+};
+
 export default function WebexActivityStream(props) {
   const {roomID, adapters} = props;
   const {roomsAdapter, activitiesAdapter, peopleAdapter} = adapters;
   const {title, roomType} = useRoom(roomID, roomsAdapter);
-  const activityIDs = useActivityStream(roomID, roomsAdapter);
+  const activitiesData = useActivityStream(roomID, roomsAdapter);
   const personName = roomType === RoomType.DIRECT ? title : '';
-  const activities = activityIDs.map((activityID) => (
-    <WebexActivity key={activityID} activityID={activityID} adapters={{activitiesAdapter, peopleAdapter}} />
-  ));
+  const activities = activitiesData.map((activity) => {
+    // If the activity is an object with a date property, it is a time ruler
+    const activityComponent = activity.date ? (
+      <TimeRuler key={activity.date.toString()} text={formatTimeRulerText(activity.date)} />
```

Comment:
You can avoid pre-formatting by just creating the Date object here.
```suggestion
      <TimeRuler key={activity.date.toString()} text={formatTimeRulerText(new Date(activity.date))} />
```

Reply:
I don't understand what you mean by this. `activity.date` is already a date object.

Reply:
This is based on my previous comment on the activity formatting, assuming that the adapter won't pass a `Date` object anymore